DRA pod status (KEP 4680) (#48547)
kubernetes/website: content/en/docs/concepts/scheduling-eviction/dynamic-resource-allocation.md

In addition to enabling the feature in the cluster, a resource driver also has to
be installed. Please refer to the driver's documentation for details.

## Devices status

{{< feature-state feature_gate_name="ResourceHealthStatus" >}}

There are cases when devices fail or are shut down. The responsibility of the DRA plugin
in this case is to notify the kubelet about the situation using the `WatchResources` API.

> Pods that were assigned to the failed devices will continue be assigned to this device.

Reviewer: ```suggestion
Pods that were assigned to the failed devices will continue be assigned to these devices.
```

> It is typical that code relying on the device will start failing and Pod may get
> into Failed phase if `restartPolicy` for the Pod was not `Always` or enter the crash loop
> otherwise.

Reviewer 2: ```suggestion
Code that relies on the devices will usually fail. Pods that don't set the `restartPolicy`
field to `Always` might enter the `Failed` phase. Pods that do set the `restartPolicy`
field to `Always` might enter a crash loop. 
```

Reviewer 2: Merge this into the previous paragraph to make that a single "problem statement"

> By enabling the feature gate `ResourceHealthStatus`, the field `allocatedResourcesStatus`
> will be added to each container status, within the `.status` for each Pod. The `allocatedResourcesStatus`
> field reports health information for each device assigned to the container.

Reviewer 2: ```suggestion
If you enable the `ResourceHealthStatus` feature gate, the `allocatedResourcesStatus` field
is added to each entry in the `status.containerStatuses` field of the Pod specification. The `allocatedResourcesStatus` field reports health information for each device assigned to the
container.
```

> For a failed Pod, or or where you suspect a fault, you can use this status to understand whether

Reviewer: `or or`

> For a failed Pod, or or where you suspect a fault, you can use this status to understand whether
> the Pod behavior may be associated with device failure. For example, if an accelerator is reporting
> an over-temperature event, the `allocatedResourcesStatus` field may be able to report this.

Reviewer 2: ```suggestion
For a failed Pod, or where you suspect a fault, you can use the `allocatedResourcesStatus`
field to understand whether the Pod behavior might be associated with device failure. For
example, if an accelerator is reporting an over-temperature event, the
`allocatedResourcesStatus` field might help you to identify the event.
```
